iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/DisabledMfaBucketDeletionCheck.java

```diff
@@ -40,9 +46,18 @@ protected void registerResourceConsumer() {
         AttributeSymbol mfaDelete = versioning.attribute("mfa_delete")
           .reportIf(isFalse(), MESSAGE, resource.toSecondary(MESSAGE_SECONDARY));
 
-        if (mfaDelete.isAbsent()) {
+        if (resource.provider(AWS).hasVersionLowerThan(AWS_V_4) && mfaDelete.isAbsent()) {
           versioning.report(MESSAGE, resource.toSecondary(MESSAGE_SECONDARY));
         }
       });
+
+    register("aws_s3_bucket_versioning", resource -> {
+      SecondaryLocation secondary = resource.toSecondary(MESSAGE_SECONDARY);
+        resource.block("versioning_configuration")
+          .attribute("mfa_delete")
+            .reportIf(equalTo("Disabled"), MESSAGE, secondary)
+            .reportIfAbsent(MESSAGE, secondary);
```

Comment:
The current description of [SONARIAC-416](https://jira.sonarsource.com/browse/SONARIAC-416) doesn't specify that an issue should be raised if `aws_s3_bucket_versioning.versioning_configuration.mfa_delete` is absent.